### ISSUE TITLE
BUG: `AttributeError` during using `Series.isin` in benchmark mode

### ISSUE THREAD

**Issue by prutskov** [modin-project/modin#4273]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04
- **Modin version** (`modin.__version__`):  e5e9634357e60925a5a70e56a1d4882d269f533a
- **Python version**: 3.8.11
- **Code we can use to reproduce**:

```python
import numpy as np
import modin.pandas as pd
import modin.config as cfg
cfg.BenchmarkMode.put(True)

nrows = 10

d = ["A", "B", "C", "D"]

s1 = pd.Series([d[np.random.randint(0, 4)] for _ in range(nrows)])
s2 = pd.Series(d)
mask = s2.isin(s1)

print(mask)

```

### Describe the problem

`AttributeError` is appeared during using `Series.isin` in benchmark mode only. 

### Source code / logs

<details><summary>Traceback</summary>

```bash
Traceback (most recent call last):                                                                      File "test.py", line 12, in <module>
    mask = s2.isin(s1)
  File "/localdisk/aprutsko/modin/modin/pandas/base.py", line 1532, in isin                               query_compiler=self._query_compiler.isin(values=values)
  File "/localdisk/aprutsko/modin/modin/core/dataframe/algebra/map.py", line 47, in caller
    query_compiler._modin_frame.map(                                                                    File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 114, in run_f_on_minimally_updated_metadata
    result = f(self, *args, **kwargs)                                                                   File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 1514, in map                                                                                                        new_partitions = self._partition_mgr_cls.map_partitions(self._partitions, func)
  File "/localdisk/aprutsko/modin/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py", line 57, in magic
    result_parts = f(*args, **kwargs)
  File "/localdisk/aprutsko/modin/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py", line 280, in map_partitions
    return super(PandasOnRayDataframePartitionManager, cls).map_partitions(
  File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/partitioning/partition_manager.py", line 58, in wait
    result = func(*args, **kwargs)
  File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/partitioning/partition_manager.py", line 487, in map_partitions
    preprocessed_map_func = cls.preprocess_func(map_func)
  File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/partitioning/partition_manager.py", line 114, in preprocess_func
    return cls._partition_class.preprocess_func(map_func)
  File "/localdisk/aprutsko/modin/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py", line 246, in preprocess_func
    return ray.put(func)
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/worker.py", line 1795, in put
    object_ref = worker.put_object(
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/worker.py", line 294, in put_object
    serialized_value = self.get_serialization_context().serialize(value)
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/serialization.py", line 367, in serialize
    return self._serialize_to_msgpack(value)
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/serialization.py", line 347, in _serialize_to_msgpack
    self._serialize_to_pickle5(metadata, python_objects)
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/serialization.py", line 307, in _serialize_to_pickle5
    raise e
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/serialization.py", line 303, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 73, in dumps
    cp.dump(obj)
  File "/localdisk/aprutsko/miniconda/envs/modin/lib/python3.8/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 620, in dump
    return Pickler.dump(self, obj)
  File "/localdisk/aprutsko/modin/modin/pandas/series.py", line 2550, in __reduce__
    self._query_compiler.finalize()
  File "/localdisk/aprutsko/modin/modin/core/storage_formats/pandas/query_compiler.py", line 254, in finalize
    self._modin_frame.finalize()
  File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 2817, in finalize
    self._partition_mgr_cls.finalize(self._partitions)
  File "/localdisk/aprutsko/modin/modin/core/dataframe/pandas/partitioning/partition_manager.py", line 66, in wait
    all(map(lambda partition: partition.wait() or True, partitions.flatten()))
AttributeError: 'NoneType' object has no attribute 'flatten'
```
</details>

**Comment by anmyachev** [comment]
`wait_computations_if_benchmark_mode` should used for functions that return partitions, however `finalize` doesn't do this.

https://github.com/modin-project/modin/blob/e5e9634357e60925a5a70e56a1d4882d269f533a/modin/core/dataframe/pandas/partitioning/partition_manager.py#L1286

@YarShev can you fix this?

**Comment by YarShev** [comment]
I am not sure that is the problem with finalize. Let's prioritize this for 0.15.

**Comment by anmyachev** [comment]
Fixed in #5006
